src/eredis_sync_parser.erl

```diff
@@ -281,8 +281,7 @@ buffer_append({List, Size}, Binary) ->
   end,
   {NewList, Size + byte_size(Binary)}.
 
-buffer_hd({[<<Char, _/binary>> | _], _}) -> [Char];
-buffer_hd({[], _}) -> [].
```

Comment:
Why do we need to remove this?

Reply:
Because second definition will be called never. And dialyzer error tell us about it
`
eredis_sync_parser.erl:285: The pattern 
          {[], _} can never match the type 
          {[any(), ...], number()}
`